pkg/controller/projectclaim/customresourceadapter.go

```diff
@@ -161,10 +163,65 @@ func (c *CustomResourceAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimS
 	}
 
 	c.projectClaim.Status.State = state
+	return c.StatusUpdate()
+}
+
+// StatusUpdate updates the project claim status
+func (c *CustomResourceAdapter) StatusUpdate() error {
 	err := c.client.Status().Update(context.TODO(), c.projectClaim)
 	if err != nil {
-		c.logger.Error(err, "Failed to update ProjectClaim state")
+		c.logger.Error(err, fmt.Sprintf("failed status update for %s", c.projectClaim.Name))
```

Comment:
+1 adding the name of the ProjectClaim to the error message. However, this new error message lacks the information that we tried to upgrade a ProjectClaim state, I think that can be helpful information.